### DISCUSSION TITLE
IL2CPP : Canvas.renderMode setter could not be unstripped

### DISCUSSION THREAD

**Discussion by akarnokd** [BepInEx/BepInEx#733] [Help and Q&A]
I'm modding a game that went from Unity C# to Unity IL2CPP (2023.1.9f). Followed the guides to enable modding in this new version via BE v679 from the nightly builds. Started converting my mods to this new environment.

All was dandy until I got a `System.NotSupportedException("Method unstripping failed")` for a Canvas.renderMode's setter.

```cs
var canvas = parent.AddComponent<Canvas>();
canvas.renderMode = RenderMode.ScreenSpaceOverlay; // <------------------------
```

Unfortunately, many of my mods create such extra canvasses as UI so I'm stuck. 

**Any tips?**

I'm not at liberty to share any binaries as this IL2CPP version of the game is in a very private beta test phase.

**Comment by thegu5** [comment]
I also ran into the issue - what worked for me was replacing the Il2CppInterop.Generator.dll in /BepInEx/core with a version that has this pr's changes: [https://github.com/BepInEx/Il2CppInterop/pull/108](https://github.com/BepInEx/Il2CppInterop/pull/108) ([patch](https://github.com/BepInEx/Il2CppInterop/pull/108.patch))

**Reply by akarnokd** [reply]
Yep, eventually figured out a [workaround](https://github.com/BepInEx/Il2CppInterop/pull/108#issuecomment-1867539462) but now the game went back to Mono for the better.
